sliceguard/embeddings.py

```diff
@@ -44,6 +45,23 @@ def generate_text_embeddings(
     return embeddings
 
 
+def generate_image_pred_probs_embeddings(
```

Comment:
Consider better extending the generate_image_embeddings method? Just check if supplied model is classification model and return probabilities if that is the case.